Fix LatinHypercube sampler

aeslaughter opened the pull request:
This fixes the LHS and adds improvements to the Sampler API to achieve the fix.

(closes #14830)
<!--
INCLUDE THE FOLLOWING IN THE PR DESCRIPTION
- Explain relevant design information for your change.
- Follow the [Coding Standards](http://mooseframework.org/wiki/CodeStandards/).
- Submit or improve [Test Cases](http://mooseframework.org/wiki/MooseTraining/testing/).
- Reference a specific issue, place "refs #<issue>" or "closes #<issue>" (e.g., #closes #1234).
-->

aeslaughter committed:
Fix LHS and improve Sampler APIs to accomadate necessary changes

(refs #14830)

aeslaughter committed:
Update tests for changes to LHS

(refs #14830)

aeslaughter committed:
WIP: Update examples for changes to LHS

(refs #14830)

aeslaughter commented:
@zachmprince Could you update the examples you created that used the LHS?
@grmnptr Any chance you could update your POD example?

grmnptr commented:
Of course! However, it will take me till Sunday because I am out of town today and tomorrow and I don't have MOOSE installed on my laptop.

aeslaughter committed:
Update LHS documentation

(refs #14830)

aeslaughter committed:
Move shuffle function into framework to allow it in Sampler objects

(refs #14830)

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/670460/) on 6bf8b7c wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/16838/site)

This comment will be updated on new commits.

grmnptr committed:
Added corrected polynomial regression example tests. (refs #14830)

Fixed POD-RB example tests with new LHS sampler. (refs #14830)

Fixed combined example tests. (refs #14830)

Updated  nonlin. param. study example md file. (refs #14830)

Adjusted example text and deleted unnecessary figure for polynomial regression. (refs #14830)

Corrected messed up sentence in polyreg example. (refs #14830)

Updated convergence and ev files for PODRB example. (refs #14830)

Update POD-RB example after LHS restructure, update large_media. (refs #14830)

Upadted POD-RB md file to match numbers on figure. (refs #14830)

Fixed combined transient surrogate example results after LHS restructure. (#refs 14830)

zachmprince committed:
Fixing thermomechanics example with new LHS #14830

aeslaughter committed:
Small fixes due to rebase

(refs #14830)

aeslaughter committed:
Update gold for changes to output format and example content

(refs #14830)

aeslaughter committed:
Add necessary python packages to test spec and limit util tests in recover

(refs #14830)

aeslaughter commented:
@zachmprince This is ready to go, thanks for fixing my force-push mistake! And thanks to you and @grmnptr for getting this all fixed up.
